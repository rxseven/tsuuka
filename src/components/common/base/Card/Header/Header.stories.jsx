import React from 'react';
import { storiesOf } from '@storybook/react';

import Header from './index';

const title = 'Components/Common/Base/Card/Header';
const component = <Header>Card Header</Header>;

storiesOf(title, module).add('default', () => component);
