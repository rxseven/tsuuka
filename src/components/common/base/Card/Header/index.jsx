import PropTypes from 'prop-types';
import exact from 'prop-types-exact';
import React from 'react';

const propTypes = exact({
  children: PropTypes.node.isRequired
});

function Header({ children }) {
  return <div className="card-header">{children}</div>;
}

Header.propTypes = propTypes;

export default Header;
