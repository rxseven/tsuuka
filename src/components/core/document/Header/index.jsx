import React from 'react';
import styled from 'styled-components';

import Grid from 'components/common/base/Grid';
import Brand from 'components/common/base/Brand';
import Burger from 'components/common/base/Burger';
import GitHub from 'components/common/composite/GitHub';
import withContext from 'HOCs/withContext';

const Content = styled.div`
  align-items: center;
  display: flex;
  height: 30px;
  justify-content: space-between;
  margin: 0.5rem 0;
`;

const Frame = styled.header`
  background-color: #fff;
  box-shadow: 0 3px 4px 0 rgba(0, 0, 0, 0.2), 0 3px 3px -2px rgba(0, 0, 0, 0.14),
    0 1px 8px 0 rgba(0, 0, 0, 0.12);
  flex-shrink: 0;
  left: 0;
  padding: 0;
  position: fixed;
  right: 0;
  top: 0;
  width: 100%;
  z-index: 2;
`;

const Meta = styled.div`
  align-items: center;
  display: flex;
`;

const Nav = styled.div`
  align-items: center;
  display: flex;
`;

const Toggle = styled(Burger)`
  margin-left: -0.65rem;
  margin-right: 0.75rem;
`;

function Header({ context }) {
  const { actions } = context;
  const { onToggle } = actions.menu;

  return (
    <Frame>
      <Grid.Container>
        <Grid.Row>
          <Grid.Column>
            <Content>
              <Nav>
                <Toggle onToggle={onToggle} />
                <Brand />
              </Nav>
              <Meta>
                <GitHub />
              </Meta>
            </Content>
          </Grid.Column>
        </Grid.Row>
      </Grid.Container>
    </Frame>
  );
}

export default withContext(Header);
