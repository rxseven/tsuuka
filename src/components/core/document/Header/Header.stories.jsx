import React from 'react';
import { storiesOf } from '@storybook/react';

import Header from './index';

const title = 'Components/Core/Document/Header';
const component = <Header />;

storiesOf(title, module).add('default', () => component);
