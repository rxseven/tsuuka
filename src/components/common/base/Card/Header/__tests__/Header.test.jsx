import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Header from '../index';

describe('<Card.Header />', () => {
  // Arrange
  const props = { children: 'content' };
  const component = <Header {...props} />;

  describe('Unit tests', () => {
    it('should render without crashing', () => {
      // Act
      const wrapper = shallow(component);

      // Assert
      expect(wrapper).toBeDefined();
    });
  });

  describe('Snapshot tests', () => {
    it('should render correctly', () => {
      // Act
      const tree = renderer.create(component).toJSON();

      // Assert
      expect(tree).toMatchSnapshot();
    });
  });
});
