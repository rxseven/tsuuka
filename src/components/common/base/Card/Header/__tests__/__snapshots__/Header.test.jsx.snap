// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card.Header /> Snapshot tests should render correctly 1`] = `
<div
  className="card-header"
>
  content
</div>
`;
